bentoml/handlers/tensorflow_tensor_handler.py

```diff
@@ -16,19 +16,163 @@
 from __future__ import division
 from __future__ import print_function
 
-from bentoml.handlers.base_handlers import BentoHandler
+import json
+import tensorflow as tf
+from flask import make_response, Response, jsonify
+from bentoml.handlers.base_handlers import BentoHandler, get_output_str
+
+
+B64_KEY = 'b64'
+
+def decode_b64_if_needed(value):
+    if isinstance(value, dict):
+        if B64_KEY in value:
+            return base64.b64decode(value[B64_KEY])
+        else:
+            new_value = {}
+            for k, v in value.iteritems():
+                new_value[k] = decode_b64_if_needed(v)
+            return new_value
+    elif isinstance(value, list):
+        new_value = []
+        for v in value:
+            new_value.append(decode_b64_if_needed(v))
+        return new_value
+    else:
+        return value
 
 
 class TensorflowTensorHandler(BentoHandler):
     """
     Tensor handlers for Tensorflow models
+    Transform incoming tf tensor data from http request, cli or lambda event into tf tensor
+    The behaviour should be compatible with tensorflow serving REST predict API: 
+        * https://www.tensorflow.org/tfx/serving/api_rest#predict_api
+
+    Args:
+        dtype (tf.dtypes): The expected dtype of the input tensor
+        shape (tuple): The expected shape of the input tensor.
+            shapes like (None, 28, 28) are supported
+
+    Raises:
+        BentoMLException: BentoML currently doesn't support Content-Type
     """
 
+    def __init__(self, spec=None):
+        self.spec = spec
+        #self.input_names = input_names
+
+    @property
+    def request_schema(self):
+
+        return {
+            "application/json": {
+                "schema": {
+                    "type": "object",
+                    "properties": {
+                        "signature_name": {
+                            "type": "string",
+                            "default": None,
+                        },
+                        "instances": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                            },
+                            "default": None,
+                        },
+                        "inputs": {
+                            "type": "object",
+                            "default": None,
+                        }
+                    },
+                }
+            }
+        }
+
+    def _handle_raw_str(self, raw_str, output_format, func):
+        parsed_json = json.loads(raw_str)
+        if parsed_json.get("instances") is not None:
+            instances = parsed_json.get("instances")
+            instances = decode_b64_if_needed(instances)
+
+            if self.spec is not None:
+                parsed_tensor = tf.constant(instances, self.spec.dtype)
+                # origin_shape_map = {parsed_tensor._id: parsed_tensor.shape}
```

Comment:
Why this comment: https://www.notion.so/serving-1a05ac2a425c450e857b65c8b401aee6